bodhi/tests/server/test_validators.py

```diff
@@ -907,3 +907,25 @@ def test_unknown_koji_genericerror(self):
 
         # check type of wrapped exception
         assert isinstance(excinfo.value.__cause__, koji.GenericError)
+
+    def test_with_unknown_tag(self):
+        """Test to prevent users to create an update from a tag that doesn't exist"""
+        with mock.patch('bodhi.server.validators.buildsys.get_session',
+                        self.mock_get_session_for_class(self.UnknownTagDevBuildsys)):
+            validators.validate_from_tag(self.request)
+
+        assert self.request.errors == [
+            {'location': 'body', 'name': 'from_tag',
+             'description': "The supplied from_tag doesn't exist."}
+        ]
+
+    def test_without_sidetag(self):
+        """A tag that is not a sidetag should add an error to the request"""
+        with mock.patch('bodhi.server.validators.buildsys.get_session',
+                        self.mock_get_session_for_class(self.UnknownTagDevBuildsys)):
+            validators.validate_from_tag(self.request)
```

Comment:
This test shouldn't mock the Koji client with one which always behaves as if the tag is unknown -- it is about known tags which aren't side tags. I.e., it should look like this:

````
    def test_without_sidetag(self):
        """A tag that is not a sidetag should add an error to the request"""
        self.request.validated['from_tag'] = 'no-side-tag'

        validators.validate_from_tag(self.request)

        assert self.request.errors == [ 
            {'location': 'body', 'name': 'from_tag',
             'description': "The supplied tag is not a side tag."}
        ]
````

For this to work, the mocked Koji client (from `bodhi/server/buildsys.py`) needs to return the `extra` dict which we assume above, or the code needs to cope with it missing from the API response. To be safe, we should do both :wink::

````
diff --git a/bodhi/server/buildsys.py b/bodhi/server/buildsys.py
index 6e9048bc..dd198f9d 100644
--- a/bodhi/server/buildsys.py
+++ b/bodhi/server/buildsys.py
@@ -464,7 +464,7 @@ class DevBuildsys:
                     'id': 7777}
 
         return {'maven_support': False, 'locked': False, 'name': taginfo,
-                'perm': None, 'id': 246, 'arches': None,
+                'extra': {}, 'perm': None, 'id': 246, 'arches': None,
                 'maven_include_all': False, 'perm_id': None}
 
     def getFullInheritance(self, taginfo, **kw):
diff --git a/bodhi/server/validators.py b/bodhi/server/validators.py
index c96919b4..389cf8e9 100644
--- a/bodhi/server/validators.py
+++ b/bodhi/server/validators.py
@@ -1328,7 +1328,7 @@ def validate_from_tag(request: pyramid.request.Request, **kwargs: dict):
             return
 
         # prevent user from creating an update if tag is not a side tag
-        if not taginfo['extra'].get('sidetag'):
+        if not taginfo.get('extra', {}).get('sidetag'):
             request.errors.add('body', 'from_tag', "The supplied tag is not a side tag.")
             return
 
````